cmd/ef_tests/levm/runner/levm_runner.rs

```diff
@@ -78,7 +78,7 @@ pub fn prepare_vm_for_tx(vector: &TestVector, test: &EFTest) -> Result<VM, EFTes
             origin: test.transactions.get(vector).unwrap().sender,
             consumed_gas: U256::default(),
             refunded_gas: U256::default(),
-            gas_limit: test.env.current_gas_limit, //this should be tx gas limit
+            gas_limit: test.transactions.get(vector).unwrap().gas_limit,
```

Comment:
Return an internal error instead of unwrapping here.

Reply:
Maybe do:
```rust
let tx = test.transaction.get(vector).ok_or(...)
```
and then use tx in every single place that needs it.

Reply:
If this behavior is repeated elsewhere, it should be handled also.

Reply:
Done in [Return EFTestRunnerError instead of using unwrap](https://github.com/lambdaclass/ethrex/pull/1281/commits/06d25f210a3480648e87b8f061f0dd89e84bbac6)